src/main.cpp

```diff
@@ -33,63 +37,136 @@ constexpr auto VERSION = _VERSION;
+ private:
+  std::vector<std::string> tokens;
+};
+
+// Parse the command line arguments
+void parseArgs(int argc, char** argv, bool& daemonMode, bool& clientMode,
+               double& startThreshold, double& finishThreshold) {
+  bool verbose, quiet, noGestures, noUpdates;
+
+  InputParser input(argc, argv);
+
+  // daemon takes precedence over client
+  if (input.cmdOptionExists("--daemon")) {
+    daemonMode = true;
+    input.getCmdOption2d("--daemon", startThreshold, finishThreshold);
+  } else if (input.cmdOptionExists("--client") || (argc == 1)) {
+    clientMode = true;
+  }
+
+  verbose = input.cmdOptionExists("--verbose") || input.cmdOptionExists("-v");
+  quiet = input.cmdOptionExists("--quiet") || input.cmdOptionExists("-q");
+  noGestures = input.cmdOptionExists("--no-gesture-messages");
+  noUpdates = input.cmdOptionExists("--no-update-messages");
+
+  // init Logger options
+  Logger::obj(verbose, quiet, noGestures, noUpdates);
 }
 
-int main(int argc, char **argv) {
-  printWelcomeMessage();
+void printWelcomeMessage() {
+  Logger& log = Logger::obj();
+  log.info << "Touchégg " << VERSION << "." << std::endl;
+  log.info << "Usage: touchegg [--verbose | -v] [--quiet | -q] "
+              "[--no-gesture-messages] [--no-update-messages] "
+              "[--daemon [start_threshold finish_threshold]] "
+              "[--client]"
+           << std::endl
+           << std::endl;
+
+  log.info << "Multi-touch gesture recognizer." << std::endl;
+  log.info << "Touchégg is an app that runs in the background and transforms "
+              "the gestures you make on your touchpad into visible actions in "
+              "your desktop."
+           << std::endl;
+  log.info << "For more information please visit:" << std::endl;
+  log.info << "https://github.com/JoseExposito/touchegg" << std::endl
+           << std::endl;
+
+  log.info << "Option\t\tMeaning" << std::endl;
+  log.info << " --daemon\tRun Touchégg in daemon mode. This mode starts a "
+              "service that gathers gestures but executes no actions"
+           << std::endl;
+  log.info << " --client\tConnect to an existing Touchégg daemon and "
+              "execute actions in your desktop"
+           << std::endl;
+  log.info << "Without arguments Touchégg starts in client mode" << std::endl
+           << std::endl;
+}
 
-  // Parse the command line arguments
+int main(int argc, char** argv) {
   bool daemonMode = false;
-  bool clientMode = (argc == 1);
+  bool clientMode = false;
   double startThreshold = -1;
   double finishThreshold = -1;
 
-  if (argc > 1) {
-    std::string param{argv[1]};  // NOLINT
-    daemonMode = (param == "--daemon");
-    clientMode = (param == "--client");
+  parseArgs(argc, argv, daemonMode, clientMode, startThreshold,
+            finishThreshold);
+
+  /**
+   * if this convenience var is used in main() it MUST be declared after
+   * parseArgs() so that Logger options are set properly!
+   */
+  Logger& log = Logger::obj();
```

Comment:
If we initialize the logger here accessing the flags stored by the arg parser, we remove the dependency between the both of them and we can remove this "warning" comment.

Reply:
Sure, but the parser would be returning these flags not storing them. (same as the run modes)